bridges/ethMultiversX/bridgeExecutor_test.go

```diff
@@ -1006,12 +1009,151 @@ func TestMultiversXToEthBridgeExecutor_GetAndStoreBatchFromMultiversX(t *testing
 		executor, _ := NewBridgeExecutor(args)
 		batch, err := executor.GetBatchFromMultiversX(context.Background())
 		assert.True(t, wasCalled)
-		assert.Equal(t, providedBatch, batch)
+		assert.Nil(t, batch)
+		assert.Equal(t, fmt.Errorf("%w, fetched nonce: %d, num deposits: %d", ErrFinalBatchNotFound, 0, 0), err)
+	})
+	t.Run("should work", func(t *testing.T) {
+		t.Parallel()
+
+		wasPendingBatchCalled := false
+		wasGetBatchSCMetadataCalled := false
+		providedBatchWithDeposits := &bridgeCore.TransferBatch{
+			Deposits: []*bridgeCore.DepositTransfer{{}, {}},
+		}
+		args := createMockExecutorArgs()
+		args.MultiversXClient = &bridgeTests.MultiversXClientStub{
+			GetPendingBatchCalled: func(ctx context.Context) (*bridgeCore.TransferBatch, error) {
+				wasPendingBatchCalled = true
+				return providedBatchWithDeposits, nil
+			},
+			GetBatchSCMetadataCalled: func(ctx context.Context, batch *bridgeCore.TransferBatch) ([]*transaction.Events, error) {
+				wasGetBatchSCMetadataCalled = true
+				return providedTxnEvents, nil
+			},
+		}
+
+		executor, _ := NewBridgeExecutor(args)
+		batch, err := executor.GetBatchFromMultiversX(context.Background())
+		assert.True(t, wasPendingBatchCalled)
+		assert.True(t, wasGetBatchSCMetadataCalled)
+		assert.Equal(t, providedBatchWithDeposits, batch)
+		assert.Nil(t, err)
+
+		err = executor.StoreBatchFromMultiversX(batch)
+		assert.Equal(t, providedBatchWithDeposits, executor.batch)
+		assert.Nil(t, err)
+	})
+	t.Run("should add deposits metadata for sc calls", func(t *testing.T) {
+		t.Parallel()
+
+		providedNonce := uint64(8346)
+		depositNonce := uint64(100)
+		depositData := []byte("testData")
+		expectedBatch := &bridgeCore.TransferBatch{
+			ID: providedNonce,
+			Deposits: []*bridgeCore.DepositTransfer{
+				{
+					Nonce: depositNonce,
+				},
+			},
+		}
+		args := createMockExecutorArgs()
+		args.MultiversXClient = &bridgeTests.MultiversXClientStub{
+			GetPendingBatchCalled: func(ctx context.Context) (*bridgeCore.TransferBatch, error) {
+				return expectedBatch, nil
+			},
+			GetBatchSCMetadataCalled: func(ctx context.Context, batch *bridgeCore.TransferBatch) ([]*transaction.Events, error) {
+				depositNonceBytes := make([]byte, 8)
+				binary.BigEndian.PutUint64(depositNonceBytes, depositNonce)
+				return []*transaction.Events{
+					{Topics: [][]byte{{}, depositNonceBytes, {}, {}, {}, {}, {}, depositData}},
+				}, nil
+			},
+		}
+
+		executor, _ := NewBridgeExecutor(args)
+		batch, err := executor.GetBatchFromMultiversX(context.Background())
+		assert.Nil(t, err)
+		assert.Equal(t, expectedBatch, batch)
```

Comment:
why does this assert pass? The batch should contain the SC call data, the expectedBatch does not contain the SC call data. Please refactor these checks as suggested in the other comment